opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/DefaultProvisionService.java

```diff
@@ -262,7 +262,7 @@ public void updateNode(final OnmsNode node, String rescanExisting) {
         node.setMetaData(dbNode.getMetaData());
         for(final OnmsIpInterface onmsIpInterface : dbNode.getIpInterfaces()) {
             final InetAddress inetAddress = onmsIpInterface.getIpAddress();
-            if (inetAddress != null) {
+            if (inetAddress != null && node.getIpInterfaceByIpAddress(inetAddress) != null) {
```

Comment:
Should probably pull the results of `node.getIpInterfaceByIpAddress(inetAddress)` out into a variable so it doesn't have to iterate the interfaces twice.

Reply:
Yes, sure. Makes sense.